### PR TITLE
6612: Remove deprecated sync modes and related helper methods

### PR DESCRIPTION
**Pull request by Matilda-Clerke**
## PR description
Remove deprecated sync modes and related helper methods

## Fixed Issue(s)

https://github.com/hyperledger/besu/issues/6612

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [x] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

**Comment by Matilda-Clerke** [comment]
Relevant docs needing changing: https://besu.hyperledger.org/public-networks/how-to/bonsai-limit-trie-logs#unable-to-change-the-sync-mode

**Review by pinges — APPROVED** [review]
LGTM

File: besu/src/main/java/org/hyperledger/besu/controller/BesuController.java [hunk 1]
```diff
@@ -361,7 +359,7 @@ public BesuControllerBuilder fromGenesisFile(
       // wrap with TransitionBesuControllerBuilder if we have a terminal total difficulty:
       if (configOptions.getTerminalTotalDifficulty().isPresent()) {
         // Enable start with vanilla MergeBesuControllerBuilder for PoS checkpoint block
-        if (isCheckpointSync(syncMode) && isCheckpointPoSBlock(configOptions)) {
+        if (syncMode == SyncMode.CHECKPOINT && isCheckpointPoSBlock(configOptions)) {
```
Comment (siladu) [inline besu/src/main/java/org/hyperledger/besu/controller/BesuController.java:362]:
```suggestion
        if (SyncMode.CHECKPOINT == syncMode  && isCheckpointPoSBlock(configOptions)) {
```

nit: this way around avoids a null pointer if syncMode were ever to be null (which shouldn't happen to be honest)

**Review by jframe — COMMENTED** [review]
Can you add an entry in the changelog, in the breaking changes section mentioning that the X_SNAP and X_CHECKPOINT options have now been removed and to use SNAP and CHECKPOINT instead

File: besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java
Comment (siladu) [inline besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java:1]: nit: same point about the NullPointers

**Review by siladu — APPROVED** [review]
LGTM - just a non-blocking comment about NPEs

**Comment by pinges** [comment]
I guess a CHANGELOG entry would be good!!

**Comment by siladu** [comment]
> I guess a CHANGELOG entry would be good!!

yeh, that's fine to do in another PR.

**Comment by macfarla** [comment]
I think the docs changes have already been done for this, if so can remove the label

File: besu/src/main/java/org/hyperledger/besu/controller/BesuController.java (hunk 1 above)
Comment (Matilda-Clerke) [reply besu/src/main/java/org/hyperledger/besu/controller/BesuController.java:362]: I thought if syncMode were null, it should just evaluate `null == SyncMode.CHECKPOINT`, which should be ok and return false as expected, right?

Comment (siladu) [reply besu/src/main/java/org/hyperledger/besu/controller/BesuController.java:362]: Sorry, I was getting confused with `nullableField.equals(value)` idiom throwing a null pointer. 
I should have suggested SyncMode.CHECKPOINT.equals(syncMode). 

You're right, both will return false so it's not a big deal, mostly stylistic.

**Comment by bgravenorst** [comment]
Doc changes implemented here: https://github.com/hyperledger/besu-docs/commit/21271eb8341a8f1572427d4ddf72322befc446c9

**Pull request merged**